sympy/polys/domains/mpelements.py

```diff
@@ -22,10 +22,7 @@ def _set_mpf(self, val):
         prec, rounding = self.context._prec_rounding
         tol = self.context.tol
 
-        if mpf_lt(mpf_abs(val, prec, rounding), tol):
-            self.__mpf__ = fzero
-        else:
-            self.__mpf__ = val
+        self.__mpf__ = val
```

Comment:
@mattpap how can I test that the polynomial division is not broken? Because all tests pass with my change.